packages/sdk/src/AbstractConfigManager.ts

```diff
@@ -0,0 +1,654 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { readFileSync } from "node:fs";
+import * as path from "node:path";
+import { ProfileConstants } from "@zowe/core-for-zowe-sdk";
+import {
+    Config,
+    ConfigBuilder,
+    ConfigSchema,
+    ConfigUtils,
+    type IConfig,
+    type IConfigBuilderOpts,
+    type IImperativeConfig,
+    type IProfAttrs,
+    type IProfile,
+    type IProfileLoaded,
+    type IProfileTypeConfiguration,
+    type ProfileInfo,
+} from "@zowe/imperative";
+import { Client, type ClientChannel } from "ssh2";
+import { type ISshConfigExt, ZClientUtils } from "./ZClientUtils";
+import { MESSAGE_TYPE, type inputBoxOpts, type qpItem, type qpOpts } from "./doc";
+
+export type ProgressCallback = (percent: number) => void;
+export abstract class AbstractConfigManager {
+    public constructor(private mProfilesCache: ProfileInfo) {}
+
+    protected abstract showMessage(message: string, type: MESSAGE_TYPE): void;
+    protected abstract showInputBox(opts: inputBoxOpts): Promise<string | undefined>;
+    protected abstract withProgress<T>(message: string, task: (progress: ProgressCallback) => Promise<T>): Promise<T>;
+    protected abstract showQuickPick(opts: qpOpts): Promise<string | undefined>;
+    protected abstract showCustomQuickPick(opts: qpOpts): Promise<qpItem | undefined>;
```

Comment:
The term "quick pick" is specific to VS Code. Should we use a more generic name in the abstract class like "menu"?